opentelemetry/benches/context_attach.rs

```diff
@@ -0,0 +1,122 @@
+use criterion::{
+    black_box, criterion_group, criterion_main, measurement::WallTime, BenchmarkGroup, BenchmarkId,
+    Criterion, Throughput,
+};
+use opentelemetry::{
+    trace::{SpanContext, TraceContextExt},
+    Context,
+};
+
+// Run this benchmark with:
+// cargo bench --bench current_context
+
+fn criterion_benchmark(c: &mut Criterion) {
+    let span_context = Context::new().with_remote_span_context(SpanContext::empty_context());
+    let contexts = vec![
+        ("empty_cx", Context::new()),
+        ("single_value_cx", Context::new().with_value(Value(4711))),
+        ("span_cx", span_context),
+    ];
+    for (name, cx) in contexts {
+        single_cx_scope(&mut group(c), name, &cx);
+        nested_cx_scope(&mut group(c), name, &cx);
+        overlapping_cx_scope(&mut group(c), name, &cx);
+    }
+}
+
+fn single_cx_scope(
+    group: &mut BenchmarkGroup<'_, WallTime>,
+    context_type: &str,
+    context: &Context,
+) {
+    let _restore = Context::current().attach();
+    group.throughput(Throughput::Elements(1)).bench_function(
+        BenchmarkId::new("single_cx_scope", context_type),
+        |b| {
+            b.iter_batched(
+                || context.clone(),
+                |cx| {
+                    single_cx(cx);
+                },
+                criterion::BatchSize::SmallInput,
+            );
+        },
+    );
+}
+
+#[inline(never)]
+fn single_cx(cx: Context) {
+    let cx = black_box(cx.attach());
+    let _ = black_box(dummy_work());
+    drop(cx);
```

Comment:
is it intentional to do explicit drop of `cx`? Any reason why it is done so?

Reply:
The `drop` will happen anyway, so I can remove it.